File: benchmarks/pandas/bench_readParquet.py
"""
Benchmark: read_parquet / to_parquet — Parquet round-trip on 10k rows
"""
import json
import time
import io
import pandas as pd
import numpy as np

ROWS = 10_000
WARMUP = 3
ITERATIONS = 20

rng = np.random.default_rng(42)
df = pd.DataFrame({
    "id": np.arange(ROWS, dtype=np.int64),
    "value": np.arange(ROWS, dtype=np.float64) * 1.1,
    "label": [f"cat_{i % 50}" for i in range(ROWS)],
})

# Warm up
for _ in range(WARMUP):
    buf = io.BytesIO()
    df.to_parquet(buf)
    buf.seek(0)
    pd.read_parquet(buf)

# Measure round-trip
start = time.perf_counter()
for _ in range(ITERATIONS):
    buf = io.BytesIO()
    df.to_parquet(buf)
    buf.seek(0)
    pd.read_parquet(buf)
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "readParquet",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_readParquet.ts
/**
 * Benchmark: readParquet / toParquet — Parquet round-trip on 10k rows
 */
import { DataFrame, toParquet, readParquet } from "../../src/index.js";

const ROWS = 10_000;
const WARMUP = 3;
const ITERATIONS = 20;

// Build a DataFrame with int, float, and string columns
const ids = Array.from({ length: ROWS }, (_, i) => i);
const values = Array.from({ length: ROWS }, (_, i) => i * 1.1);
const labels = Array.from({ length: ROWS }, (_, i) => `cat_${i % 50}`);

const df = new DataFrame({ id: ids, value: values, label: labels });

// Warm up
for (let i = 0; i < WARMUP; i++) {
  const buf = toParquet(df);
  readParquet(buf);
}

// Measure round-trip
const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  const buf = toParquet(df);
  readParquet(buf);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "readParquet",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
